Дипломный проект. Этап 1. Сервис статистики.

ViktorSalk opened the pull request:
1) Создал ветку stat_svc.
2) Подготовил многомодульность.

Viktor3642 committed:
Дипломный проект. Этап 1. Сервис статистики.
1) Создал ветку stat_svc.
2) Подготовил многомодульность.

Viktor3642 committed:
Дипломный проект. Этап 1. Сервис статистики.
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентская часть и серверная часть: stat_client и stat_server.
4) Выделил третий подмодуль, с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через docker-compose.yml.

Viktor3642 committed:
Дипломный проект. Этап 1. Сервис статистики.
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентская часть и серверная часть: stat_client и stat_server.
4) Выделил третий подмодуль, с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через docker-compose.yml.
6) Добавил postman.

Viktor3642 committed:
Дипломный проект. Этап 1. Сервис статистики.
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентская часть и серверная часть: stat_client и stat_server.
4) Выделил третий подмодуль, с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через docker-compose.yml.
6) Добавил postman, обновил pom и gitignore.

Viktor3642 committed:
Дипломный проект. Этап 1. Сервис статистики.
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентская часть и серверная часть: stat_client и stat_server.
4) Выделил третий подмодуль, с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через docker-compose.yml.
6) Добавил postman, обновил pom и gitignore.

Viktor3642 committed:
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентскую часть и серверную часть: stat_client и stat_server.
4) Выделил третий подмодуль с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через файл docker-compose.yml.
6) Добавил postman, обновил pom и gitignore.

Viktor3642 committed:
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентскую часть и серверную часть: stat_client и stat_server.
4) Выделил третий подмодуль с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через файл docker-compose.yml.
6) Добавил postman, обновил pom и gitignore.

Viktor3642 committed:
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентскую часть и серверную часть: stat_client и stat_server.
4) Выделил третий подмодуль с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через файл docker-compose.yml.
6) Добавил postman, обновил pom и gitignore.
7) Добавил тесты.

VadimZharkov commented on ewm-stat-service/pom.xml:
```diff
@@ -0,0 +1,26 @@
+        <module>stat_server</module>
+    </modules>
+
+    <properties>
```
properties уже есть в родительском pom, из него "занаследуется". в дочерних pom прописывать избыточно.

ViktorSalk replied:
Спасибо) Удалил избыточные свойства из дочернего Pom.

VadimZharkov commented on ewm-stat-service/stat_server/src/main/java/ru/practicum/model/Stat.java:
```diff
@@ -0,0 +1,40 @@
+
+import java.time.LocalDateTime;
+
+@Data
```
`@Data` для сущностей в дипломе не рекомендуется использовать.
https://habr.com/ru/companies/haulmont/articles/564682/

ViktorSalk replied:
Удалил аннотации @Data, @EqualsAndHashCode заменив на @Getter, @Setter, @ToString и явные реализации методов  equals() и hashCode().

VadimZharkov reviewed (changes requested):
Хорошая работа, структура проекта реализована верно. Нужно исправить пару недочетов.

Viktor3642 committed:
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентскую часть и серверную часть: stat_client и stat_server.
4) Выделил третий подмодуль с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через файл docker-compose.yml.
6) Добавил postman, обновил pom и gitignore.
7) Добавил тесты.
8) Удалил избыточные свойства из дочернего Pom.

Viktor3642 committed:
Дипломный проект. Этап 1. Сервис статистики.
1) Создал ветку stat_svc.
2) Подготовил многомодульность:
Основной сервис ewm-main-service и Сервис статистики ewm-stat-service.
3) Разделил Сервис на два подмодуля: клиентскую часть и серверную часть: stat_client и stat_server.
4) Выделил третий подмодуль с DTO-объектами для передачи данных.
5) Проект настроен для работы с Docker через файл docker-compose.yml.
6) Добавил postman, обновил pom и gitignore.
7) Добавил тесты.
8) Удалил избыточные свойства из дочернего Pom.
9) Удалил аннотации @Data, @EqualsAndHashCode заменив на @Getter, @Setter, @ToString и явные реализации методов  equals() и hashCode().
